src/main/java/sg/ncl/MainController.java

```diff
@@ -1918,6 +1950,106 @@ public String validateExperiment(
+        model.addAttribute("singleNodeInfoMap", singleNodeInfoMap);
+        model.addAttribute("pathTeamId", teamId);
+        model.addAttribute("pathExperimentId", expId);
+        model.addAttribute("pathNodeId", nodeId);
+        model.addAttribute("experimentName", realization.getExperimentName());
+        model.addAttribute("saveImageForm", saveImageForm);
+        return "save_experiment_image";
+    }
+
+    // bindingResult is required in the method signature to perform the JSR303 validation for Image object
+    @RequestMapping(value = "/experiments/save_image/{teamId}/{expId}/{nodeId}", method = RequestMethod.POST)
+    public String saveExperimentImage(
+            @Valid @ModelAttribute("saveImageForm") Image saveImageForm,
+            BindingResult bindingResult,
+            RedirectAttributes redirectAttributes,
+            @PathVariable String teamId,
+            @PathVariable String expId,
+            @PathVariable String nodeId) throws WebServiceRuntimeException {
+
+        if (saveImageForm.getImageName().length() < 2) {
+            log.info("Save image form has errors {}", saveImageForm);
+            redirectAttributes.addFlashAttribute("message", "Image Name minimum 2 characters");
+            return "redirect:/experiments/save_image/" + teamId + "/" + expId + "/"  + nodeId;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        HttpEntity<String> request;
+
+        try {
+            request = createHttpEntityWithBody(mapper.writeValueAsString(saveImageForm));
+        } catch (JsonProcessingException e) {
+            log.warn("JsonProcessingException {}", e);
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.saveImage(), HttpMethod.POST, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            return processSaveImageRequest(saveImageForm, redirectAttributes, teamId, expId, nodeId, response, responseBody);
+        } catch (IOException ioe) {
+            log.warn("IOException {}", ioe);
+            throw new WebServiceRuntimeException(ioe.getMessage());
+        }
+    }
+
+    private String processSaveImageRequest(@Valid @ModelAttribute("saveImageForm") Image saveImageForm, RedirectAttributes redirectAttributes, @PathVariable String teamId, @PathVariable String expId, @PathVariable String nodeId, ResponseEntity response, String responseBody) throws IOException {
+        if (RestUtil.isError(response.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+            log.warn("Save image exception: {}", exceptionState);
+
+            switch (exceptionState) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)